src/internal/operators/map.ts

```diff
@@ -43,49 +42,39 @@ import { lift } from '../util/lift';
  * @name map
  */
 export function map<T, R>(project: (value: T, index: number) => R, thisArg?: any): OperatorFunction<T, R> {
+
   return function mapOperation(source: Observable<T>): Observable<R> {
     if (typeof project !== 'function') {
       throw new TypeError('argument is not a function. Are you looking for `mapTo()`?');
     }
```

Comment:
Question. I assume this is reoccurring construct. Would an assaerFunction utility function not be easier?
```
  function asserFunction(target,errorMsg=`"${target} is not a function") {
    if (typeof target !== 'function') {
       throw new typeError(errorMsg)
   }
}
```
This will however the overall size. The downside, the call stack is a bit deeper, but even that is fixable.

Reply:
In general, I think we need to make a pass and either decide to check all of these things at runtime, or walk away from doing that and let TypeScript do its thing and let bad calls just fail to reduce bundle size. The problem is these runtime assertions are "death by 1,000 cuts" with a library like this, but on the other hand, non-TypeScript users may need them.

Reply:
> ... or walk away from doing that and let TypeScript do its thing.

That would get my vote.

Reply:
TS can't catch a thing runtime. I would say, give the dev a useful error during runtime too. 
Also, a lot of TS users will slap `:any` on anything they don't understand, nullifying all compile-time safety.
So, it's not only JS users you are serving with this. it might prevent devs from turning away from RX because they don't understand why it's erroring out.
(I'm pretty sure if you don't catch this, the error would be less then helpful)

Reply:
The scale of this is the problem, though. We'll be forcing billions of bytes over the wire for all RxJS users to accommodate developers with nicer errors when they skip type safety.

Reply:
I do realize that. But not all RxJs consumers are using TS.  Removing those assertions will add to their pain.
I see where you are coming from, and I get the point. My personal preference would really be to handle it during runtime. 
But, luckily, it is not up to me to make this decision. It's a hard one.